sql/core/src/test/resources/sql-tests/inputs/array.sql

```diff
@@ -137,3 +137,11 @@ select array_compact(array("a", "b", "c"));
 select array_compact(array(1D, null, 2D, null));
 select array_compact(array(array(1, 2, 3, null), null, array(4, null, 6)));
 select array_compact(array(null));
+
+-- function array_append
```

Comment:
would you mind adding a few cases here?

```
select array_append(array(), 1);
select array_append(array(), CAST(NULL AS String));
select array_append(array(CAST(NULL AS String)), CAST(NULL AS String));
```

Reply:
Added the tests but had to add the type cast for empty array in second one. Else it was returning null
select array_append(CAST(array() AS ARRAY<String>), CAST(NULL AS String));